src/mapnode.h

```diff
@@ -187,53 +207,97 @@ struct MapNode
 	 */
 	void getColor(const ContentFeatures &f, video::SColor *color) const;
 
-	void setLight(LightBank bank, u8 a_light, const ContentFeatures &f) noexcept;
-
-	void setLight(LightBank bank, u8 a_light, const NodeDefManager *nodemgr);
+	void setLight(LightBank bank, u8 a_light, ContentLightingFlags f) noexcept
+	{
+		// If node doesn't contain light data, ignore this
+		if (!f.has_light)
+			return;
+		if(bank == LIGHTBANK_DAY)
+		{
+			param1 &= 0xf0;
+			param1 |= a_light & 0x0f;
+		}
+		else if(bank == LIGHTBANK_NIGHT)
+		{
+			param1 &= 0x0f;
+			param1 |= (a_light & 0x0f)<<4;
+		}
+		else
+			assert("Invalid light bank" == NULL);
+	}
 
 	/**
 	 * Check if the light value for night differs from the light value for day.
 	 *
 	 * @return If the light values are equal, returns true; otherwise false
 	 */
-	bool isLightDayNightEq(const NodeDefManager *nodemgr) const;
+	bool isLightDayNightEq(ContentLightingFlags f) const
+	{
+		bool isEqual;
+
+		if (f.has_light) {
+			u8 day   = MYMAX(f.light_source, param1 & 0x0f);
+			u8 night = MYMAX(f.light_source, (param1 >> 4) & 0x0f);
+			isEqual = day == night;
+		} else {
+			isEqual = true;
+		}
 
-	u8 getLight(LightBank bank, const NodeDefManager *nodemgr) const;
+		return isEqual;
+	}
+
+
+	u8 getLight(LightBank bank, ContentLightingFlags f) const noexcept
+	{
+		// Select the brightest of [light source, propagated light]
+		u8 light;
+		if(f.has_light)
+			light = bank == LIGHTBANK_DAY ? param1 & 0x0f : (param1 >> 4) & 0x0f;
+		else
+			light = 0;
+		return MYMAX(f.light_source, light);
```

Comment:
Shorter, simpler.
```suggestion
	return MYMAX(f.light_source, getLightRaw(bank, f));
```
Also make `getLightRaw` inline.